smilerating/src/main/java/com/hsalf/smilerating/SmileRating.java

```diff
@@ -311,31 +317,33 @@ protected void onDraw(Canvas canvas) {
-                mBackgroundPaint.setColor((Integer) mColorEvaluator
-                        .evaluate(mMainSmileyTransformaFraction, mPlaceHolderCirclePaint.getColor(),
-                                (mSelectedSmile == TERRIBLE || mPreviousSmile == TERRIBLE) ? mAngryColor : mNormalColor));
-                mScaleMatrix.reset();
-                mSmilePath.computeBounds(mScaleRect, true);
-                float nonSelectedScale = mFloatEvaluator.evaluate(
-                        mInterpolator.getInterpolation(mMainSmileyTransformaFraction), getScale(NONE), 1f);
-                mScaleMatrix.setScale(nonSelectedScale, nonSelectedScale,
-                        mScaleRect.centerX(), mScaleRect.centerY());
-                mDummyDrawPah.reset();
-                mDummyDrawPah.addPath(mSmilePath, mScaleMatrix);
-
-                canvas.drawCircle(mFaceCenter.x, mFaceCenter.y,
-                        nonSelectedScale * (mHeight / 2f), mBackgroundPaint);
-                canvas.drawPath(mDummyDrawPah, mPathPaint);
-            } else {
-                canvas.drawCircle(mFaceCenter.x, mFaceCenter.y, mHeight / 2f, mBackgroundPaint);
-                canvas.drawPath(mSmilePath, mPathPaint);
-            }
+//            if (mSmileyNotSelectedPreviously) {
+//                Log.i(TAG, "Non selection");
+//                /*mPathPaint.setAlpha(Math.round(255 * mMainSmileyTransformaFraction));
+//                mBackgroundPaint.setAlpha(Math.round(255 * mMainSmileyTransformaFraction));*/
+//                mPathPaint.setColor((Integer) mColorEvaluator
+//                        .evaluate(mMainSmileyTransformaFraction, mPlaceHolderFacePaint.getColor(), mDrawingColor));
+//                mBackgroundPaint.setColor((Integer) mColorEvaluator
+//                        .evaluate(mMainSmileyTransformaFraction, mPlaceHolderCirclePaint.getColor(),
+//                                (mSelectedSmile == TERRIBLE || mPreviousSmile == TERRIBLE) ? mAngryColor : mNormalColor));
+//                mScaleMatrix.reset();
+//                mSmilePath.computeBounds(mScaleRect, true);
+//                float nonSelectedScale = mFloatEvaluator.evaluate(
+//                        mInterpolator.getInterpolation(mMainSmileyTransformaFraction), getScale(NONE), 1f);
+//                mScaleMatrix.setScale(nonSelectedScale, nonSelectedScale,
+//                        mScaleRect.centerX(), mScaleRect.centerY());
+//                mDummyDrawPah.reset();
+//                mDummyDrawPah.addPath(mSmilePath, mScaleMatrix);
+//
+//                canvas.drawCircle(mFaceCenter.x, mFaceCenter.y,
+//                        nonSelectedScale * (mHeight / 2f), mBackgroundPaint);
+//                canvas.drawPath(mDummyDrawPah, mPathPaint);
+//            } else {
+//                canvas.drawCircle(mFaceCenter.x, mFaceCenter.y, mHeight / 2f, mBackgroundPaint);
+//                canvas.drawPath(mSmilePath, mPathPaint);
+//            }
```

Comment:
Why did you comment out the `if (mSmileyNotSelectedPreviously)` code block?

It took me a little bit of digging when I was contributing to figure out what it does, so let me share my knowledge to you (if you don't already know). Basically, you want to provide a nice animation when you click on the smileys. When you have a selected smiley and you select a different one, it moves the smiley from the previous value to the new value.

However, if a smiley is not selected, then there is nowhere to start from. Thus, the variable mSmileyNotSelectedPreviously is used. Instead, it basically fades in/out the smiley instead.